ledger/complete/wal/checkpoint_v6_test.go

```diff
@@ -82,10 +89,54 @@ func createSimpleTrie(t *testing.T) []*trie.MTrie {
 	return tries
 }
 
+func randPathPayload() (ledger.Path, ledger.Payload) {
+	var path ledger.Path
+	rand.Read(path[:])
+	payload := testutils.RandomPayload(1, 100)
+	return path, *payload
+}
+
+func randNPathPayloads(n int) ([]ledger.Path, []ledger.Payload) {
+	paths := make([]ledger.Path, n)
+	payloads := make([]ledger.Payload, n)
+	for i := 0; i < n; i++ {
+		path, payload := randPathPayload()
+		paths[i] = path
+		payloads[i] = payload
+	}
+	return paths, payloads
+}
+
+func createMultipleRandomTries(t *testing.T) []*trie.MTrie {
+	tries := make([]*trie.MTrie, 0)
+	activeTrie := trie.NewEmptyMTrie()
+
+	// add tries with no shared paths
+	for i := 0; i < 100; i++ {
+		paths, payloads := randNPathPayloads(100)
+		activeTrie, _, err := trie.NewTrieWithUpdatedRegisters(activeTrie, paths, payloads, false)
```

Comment:
Need to reuse `activeTrie` inside loop if we want new trie to be created from previous trie.
```suggestion
                var err error
		activeTrie, _, err = trie.NewTrieWithUpdatedRegisters(activeTrie, paths, payloads, false)
```